code/game/objects/structures/poddoor_assembly.dm

```diff
@@ -0,0 +1,220 @@
+// poddoor assemblies
+// for building pod doors,
+// blast doors and such. features copious amounts of copypasted code from door_assembly.dm
+
+/obj/structure/poddoor_assembly
+	name = "blast door assembly"
+	icon = 'icons/obj/doors/blastdoor.dmi'
+	icon_state = "open"
+	anchored = FALSE
+	density = FALSE
+	max_integrity = 200
+	var/obj/item/electronics/airlock/electronics = null
+	var/poddoor_type = /obj/machinery/door/poddoor/preopen //type of pod dooor when completed
+	var/state = AIRLOCK_ASSEMBLY_NEEDS_WIRES //cope
+	var/base_name = "blast door"
+	var/created_name = null
+	var/previous_assembly = /obj/structure/poddoor_assembly
+	var/material_amt = 15
+	var/material_type = /obj/item/stack/sheet/plasteel
+	var/welded = FALSE // checks if the thing is welded to the floor
+
+/obj/structure/poddoor_assembly/Initialize()
+	. = ..()
+	update_icon()
+	update_name()
+
+/obj/structure/poddoor_assembly/examine(mob/user)
+	. = ..()
+	var/doorname = ""
+	if(created_name)
+		doorname = ", written on it is '[created_name]'"
+	switch(state)
+		if(AIRLOCK_ASSEMBLY_NEEDS_WIRES)
+			if(anchored)
+				. += "<span class='notice'>The anchoring bolts are <b>wrenched</b> in place, but the maintenance panel needs <i>wiring</i>.</span>"
+			else
+				. +=  "<span class='notice'>The anchoring bolts are <i>unwrenched</i>, and the assembly could be <i>welded apart</i>.</span>"
+		if(AIRLOCK_ASSEMBLY_NEEDS_ELECTRONICS)
+			. += "<span class='notice'>The maintenance panel is <b>wired</b>, but the circuit slot is <i>empty</i>.</span>"
+		if(AIRLOCK_ASSEMBLY_NEEDS_SCREWDRIVER)
+			. += "<span class='notice'>The circuit is <b>connected loosely</b> to its slot.</span>"
+	. += "<span class='notice'>The assembly [welded ? "is firmly <b>welded</b> into place" : "needs to be <i>welded</i> to the floor in order to complete it"].</span>"
+	. += "<span class='notice'>There is a small placard on the assembly[doorname].</span>"
+
+/obj/structure/poddoor_assembly/attackby(obj/item/W, mob/user, params)
+	if(istype(W, /obj/item/pen))
+		var/t = stripped_input(user, "Enter the name for the assembly.", name, created_name,MAX_NAME_LEN)
+		if(!t || (!in_range(src, usr) && loc != usr) )
+			return
+		created_name = t
+
+	else if(W.tool_behaviour == TOOL_WELDER)
+		if(!W.tool_start_check(user, amount=0))
```

Comment:
does check for 0 work?

Reply:
from what i can tell it sets the amount of fuel used to zero at the start of welding. Works as intended in testing, at least.